app/forms/openid_connect_logout_form.rb

```diff
@@ -62,13 +62,9 @@ def validate_identity
     errors.add(:id_token_hint, t('openid_connect.logout.errors.id_token_hint')) unless identity
   end
 
-  def service_provider
-    @_service_provider ||= ServiceProvider.from_issuer(identity&.service_provider)
-  end
-
   def extra_analytics_attributes
     {
-      client_id: service_provider.issuer,
+      client_id: identity.service_provider_record&.issuer,
```

Comment:
probably! one difference is that doing it this way means we make sure that the SP actually exists in the the SP table, and that the identity didn't have a garbage value in that column

I think in practice they'll align (since there's some validation that goes into making an `identity` record in the first place, and the chances that an identity would be created and the corresponding SP would be deleted should be pretty small) so let's go with the simplification.